### PR TITLE
add react-native-animations

### PR DESCRIPTION
**Pull request by mvillafuertem**
Hi, I'm creating this example to try to use Animations with React Native. But I have some troubles to understand some conversions. I get this facade thanks to converter.  Do I have to create a Js component? 

```scala
@JSImport("react-native", "Animated.View")
@js.native
object View extends TopLevel[AnimatedComponent[TypeofView]]

type AnimatedComponent[T /* <: typings.react.mod.ComponentType[_] */] = typings.react.mod.FC[
    typings.reactNative.mod.Animated.AnimatedProps[typings.react.mod.ComponentPropsWithRef[T]] ]
  

```

Original code
```javascript
      <Animated.View
        style={[
          StyleSheet.absoluteFillObject,
          {
            justifyContent: 'flex-end',
            alignItems: 'center',
            paddingBottom: 100,
          },
        ]}>
        <TouchableOpacity
          onPress={() => {}}>
          <View
            style={styles.roundButton}
          />
        </TouchableOpacity>
      </Animated.View>
```

Timer

### CODE REVIEW

**Comment by oyvindberg** [comment]
Hi there! Thanks, this looks cool :)

I see that the `Animated.View` component isn't generated in the `typings.reactNative.components` package, I think it's not there because there is another `View` component, and the component identification code does a [group by name](https://github.com/ScalablyTyped/Converter/blob/master/scalajs/src/main/scala/org/scalablytyped/converter/internal/scalajs/flavours/IdentifyReactComponents.scala#L83).

That same code has a concept of nested components, so that if `Animated` itself were a component, `Animated.View` would have been detected as a "sub-component". In this case it isn't, so this particular component unfortunately falls through the cracks.

This is rare enough that I'll let it slide for now, and consider doing something more intelligent with prefixes next time I touch that code. For now you found exactly the workaround that is needed.

**Comment by mvillafuertem** [comment]
Understood thanks.

Another thing is that I get this error when I try to convert `@expo/vector-icons`. Is there any workaround for this?

```

[error] (stImport) FromFolder(InFolder(/projects/ScalaJsReactDemos/react-native-animations/node_modules/@expo/vector-icons),TsIdentLibraryScoped(expo,vector-icons)) -> Right(Compilation failed: Class too large: typings/expoVectorIcons/expoVectorIconsStrings$, [Error] : Error while emitting typings/expoVectorIcons/expoVectorIconsStrings)

```

**Comment by oyvindberg** [comment]
Hmm, that's unfortunate. I'll check it out.

**Comment by oyvindberg** [comment]
Ugh and wow, it has 11597 different literal strings, and the `object` is 58k lines. I'll see if it's possible to just start widening to `String` after enough literals have passed through the machinery

**Comment by oyvindberg** [comment]
I didnt mean to close this by the way

**Comment by mvillafuertem** [comment]
Hi @oyvindberg,I get a `StackOverflowError` when I try to convert `react-native-reanimated` with `1.0.0-beta29.2` version. I have added a `.jvmopts` file but it doesn't work. I have also check with `1.0.0-beta30` version but this return a `java.lang.RuntimeException: Type parameter not provided`

File: react-native-animations/src/main/scala/demo/Navigation.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import scala.scalajs.js.{UndefOr, |}
+
+object Navigation {
+
+  val Drawer = createDrawerNavigator[Record[java.lang.String, js.UndefOr[js.Object]]]()
+
+  type NavigatorProps = Omit[ComponentProps[js.Function1[Props, React.Element]], reactNavigationCoreStrings.initialRouteName | reactNavigationCoreStrings.children | reactNavigationCoreStrings.screenOptions] with DefaultNavigatorOptions[DrawerNavigationOptions, Record[String, UndefOr[js.Object]]]
+  private val navigator: ComponentType[NavigatorProps] = Drawer.Navigator
+
+  private val navigatorComponent: JsFn.Component[NavigatorProps, CtorType.PropsAndChildren] = JsFnComponent[NavigatorProps, Children.Varargs](navigator.asInstanceOf[ComponentClassP[NavigatorProps]])
```
Comment (mvillafuertem) [inline react-native-animations/src/main/scala/demo/Navigation.scala:30]: I need to translate this component, but I don't know if this is the correct way. :(

```javascript
<NavigationContainer>
     <Drawer.Navigator initialRouteName="Home">
     <Drawer.Screen name="Home" component={HomeScreen} />
     <Drawer.Screen name="Notifications" component={NotificationsScreen} />
     </Drawer.Navigator>
 </NavigationContainer>
```

**Comment by oyvindberg** [comment]
Hey again @mvillafuertem 

Sorry this is taking time. I fixed the issue with `react-native-reanimated` now, and it'll be part of beta31 due soon.
I'll have a look at the rest shortly.

**Comment by mvillafuertem** [comment]
Thank you @oyvindberg for fix it. I'll wait when you have time to take a look at all the code, to continue working on this.

File: react-native-animations/src/main/scala/demo/Navigation.scala (hunk 1 above)
Comment (mvillafuertem) [reply react-native-animations/src/main/scala/demo/Navigation.scala:30]: Hi @oyvindberg, This week I've had some time to get back to scala.js code and I just saw that I have this pending. Could you help me with some hint. Thanks.

**Pull request merged**